File: apps/kotlin/src/main/kotlin/chapter02/Main.kt
package chapter02

import dataset.dataDir
import java.io.File

private const val TEST_SIZE = 0.3
private const val SEED = 0

private fun formatCounts(counts: Map<String, Int>): String = counts.entries.joinToString(", ") { (column, count) -> "$column=$count" }

fun main() {
    val csvFile = File(dataDir(), "iris.csv")
    val df = loadIris(csvFile)
    val split = prepareIris(csvFile, testSize = TEST_SIZE, seed = SEED)
    println("データ件数: ${df.rowsCount()}")
    println("欠損値の数: ${formatCounts(countMissing(df))}")
    println("訓練データ: ${split.xTrain.rowsCount()} 件, テストデータ: ${split.xTest.rowsCount()} 件")
    val missingTrain = countMissing(split.xTrain).values.sum()
    val missingTest = countMissing(split.xTest).values.sum()
    println("補完後の欠損値の数: 訓練データ $missingTrain, テストデータ $missingTest")
}

File: apps/python/lib/chapter02/__main__.py
import pandas as pd

from lib.chapter02.iris_preprocessing import (
    count_missing,
    load_iris,
    prepare_iris,
)
from lib.dataset import data_dir

TEST_SIZE = 0.3
SEED = 0


def format_counts(counts: pd.Series) -> str:
    return ", ".join(f"{column}={count}" for column, count in counts.items())


def main() -> None:
    csv_file = data_dir() / "iris.csv"
    df = load_iris(csv_file)
    split = prepare_iris(csv_file, test_size=TEST_SIZE, seed=SEED)
    print(f"データ件数: {len(df)}")
    print(f"欠損値の数: {format_counts(count_missing(df))}")
    print(f"訓練データ: {len(split.x_train)} 件, テストデータ: {len(split.x_test)} 件")
    missing_train = int(count_missing(split.x_train).sum())
    missing_test = int(count_missing(split.x_test).sum())
    print(
        f"補完後の欠損値の数: 訓練データ {missing_train}, テストデータ {missing_test}"
    )


if __name__ == "__main__":
    main()
